src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs

```diff
@@ -77,32 +86,35 @@ public IAnalysisIssue Convert(Message cFamilyIssue, string sqLanguage, ICFamilyR
                 .ToArray();
 
             // If PartsMakeFlow is set to true the issues are expected to be in the reversed order
-            if (locations != null && cFamilyIssue.PartsMakeFlow)
+            if (cFamilyIssue.PartsMakeFlow)
             {
                 Array.Reverse(locations);
             }
+
             var flows = locations.Any() ? new[] { new AnalysisIssueFlow(locations) } : null;
 
             return ToAnalysisIssue(cFamilyIssue, sqLanguage, defaultSeverity, defaultType, flows, fileContents);
         }
 
-        private IDictionary<string, string> LoadFileContentsOfReportedFiles(Message cFamilyIssue)
+        private IDictionary<string, ITextDocument> LoadFileContentsOfReportedFiles(Message cFamilyIssue)
         {
             var filePaths = cFamilyIssue.Parts
                 .Select(x => x.Filename)
                 .Union(new[] {cFamilyIssue.Filename})
                 .Distinct();
 
             return filePaths.ToDictionary(x => x,
-                path => fileSystem.File.Exists(path) ? fileSystem.File.ReadAllText(path) : null);
+                path => fileSystem.File.Exists(path)
+                    ? textDocumentFactoryService.CreateAndLoadTextDocument(path, filesContentType)
+                    : null);
```

Comment:
Worth calling out that the document is being loaded from disc, so it should match the version that was analysed by the subprocess.